backend/eventloop/application_event_loop/application_event_runner_deployments_test.go

```diff
@@ -0,0 +1,97 @@
+package application_event_loop
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Application Event Runner Deployments", func() {
+	Context("createSpecField() should generate a valid argocd Application", func() {
+		It("Input spec is converted to an argocd Application", func() {
+			input := fakeInputSpec(false, false)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplication()))
+		})
+
+		It("Sanitize illegal characters from input", func() {
+			input := fakeInputSpec(false, true)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplication()))
+		})
+
+		It("Input spec with automated enabled should set automated sync policy", func() {
+			input := fakeInputSpec(true, false)
+			application, err := createSpecField(input)
+			Expect(err).To(BeNil())
+			Expect(application).To(Equal(validApplicationWithAutomatedSync()))
+		})
+	})
+})
+
+func fakeInputSpec(automated, unsanitized bool) argoCDSpecInput {
+	input := argoCDSpecInput{
+		crName:               "sample-depl",
+		crNamespace:          "kcp-workspace",
+		destinationNamespace: "prod",
+		destinationName:      "in-cluster",
+		sourceRepoURL:        "https://github.com/test/test",
+		sourcePath:           "environments/prod",
+		automated:            automated,
+	}
+
+	if unsanitized {
+		input.sourcePath = "environments/&&prod\n"
+		input.sourceRepoURL = "https://github.com/```%test/test"
+	}
+	return input
+}
+
+func validApplicationWithAutomatedSync() string {
```

Comment:
why not create a `const` instead ?

Reply:
Yes. It makes sense to use const here. I've generalized it further to use structs in a function so we can unmarshall it when required. It avoids duplication and we no longer have to maintain raw YAML strings.